src/core/scheduler.h

```diff
@@ -74,12 +75,23 @@ class Scheduler {
   VisualizationAdaptor* visualization_ = nullptr;  //!
   RootAdaptor* root_visualization_ = nullptr;      //!
 
-  BoundSpace* bound_space_;
-  DisplacementOp* displacement_;
-  DiffusionOp* diffusion_;
-
-  std::vector<Operation*> operations_;  //!
-  std::set<std::string> protected_operations_;
+  /// List of operations that are to be added in the upcoming timestep
+  std::vector<Operation*> ops_to_add_;  //!
+  /// List of operations that are to be removed in the upcoming timestep
+  std::vector<Operation*> ops_to_remove_;  //!
+  /// List of operations that were removed from scheduling, but could be reused
+  /// later on in a simulation
+  std::vector<Operation*> unscheduled_ops_;  //!
+  /// List of operations will be executed as a stand-alone operation
+  std::vector<Operation*> scheduled_column_wise_ops_;  //!
+  /// List of operations will be executed on all simulation objects
+  std::vector<Operation*> scheduled_row_wise_ops_;  //!
```

Comment:
Can we improve the naming?

Reply:
Do you refer to the [row|column]-wise phrasing? If so, do you have any suggestions? How about:

`StandaloneOp`: an operation that is run one time per simulation step
`IterativeOp`: an operation that is run one time per simulation step **for all simulation objects**